### PR TITLE
chore: add a lint for shell script

### PR DESCRIPTION
**Pull request by JoeWang1127**
A follow up of #1961.

### CODE REVIEW

**Comment by JoeWang1127** [comment]
The lint for shell script is only triggered when there's change in `library_generation/` directory. 

To test the ci, I added a trivial change and the log is [here](https://github.com/googleapis/sdk-platform-java/actions/runs/6113885418/job/16594280379).

**Comment by suztomo** [comment]
1. Do you find anything that is different from Google shell script style? https://google.github.io/styleguide/shellguide.html (such as indent size)? If not, that's good. (No need to put effort to find discrepancy)
2. Can you add a commit in this pull request that touch some file in the "library_generation/" directory? This pull request should be merged only when the lint passes.

**Review by suztomo — APPROVED** [review]

File: .github/workflows/verify_library_generation.yaml [hunk 1]
```diff
@@ -38,3 +38,13 @@ jobs:
       run: |
         set -x
         library_generation/test/generate_library_unit_tests.sh
+  lint:
+    runs-on: ubuntu-22.04
+    steps:
+    - uses: actions/checkout@v3
+    - name: Run ShellCheck
+      uses: ludeeus/action-shellcheck@2.0.0
```
Comment (blakeli0) [inline .github/workflows/verify_library_generation.yaml:46]: @suztomo Do we have any guidelines or best practices regarding using third_party Github actions? This one has 200+ stars so there are some usages, but do we think there is any risks in using it?

Comment (suztomo) [reply .github/workflows/verify_library_generation.yaml:46]: I don't think of any risk using this because it doesn't add any code to the repository.

Comment (JoeWang1127) [reply .github/workflows/verify_library_generation.yaml:46]: The underlying [ShellCheck](https://github.com/koalaman/shellcheck) is open-sourced and recommended by Google.

If we think third-party action is unreliable, I can write a commend to invoke `shellcheck` directly. WDYT?

Comment (JoeWang1127) [reply .github/workflows/verify_library_generation.yaml:46]: I think we can use the action for now and if it's broken in the future, I can always convert it to `shellcheck` command.

Comment (meltsufin) [reply .github/workflows/verify_library_generation.yaml:46]: It would be nice to be able to run it locally too.

**Review by meltsufin — APPROVED** [review]

**Review by blakeli0 — APPROVED** [review]

**Pull request merged**